packages/astro/src/assets/utils/resolveImports.ts

```diff
@@ -0,0 +1,40 @@
+import { resolve } from 'node:path/posix';
+import { CONTENT_IMAGE_FLAG, IMAGE_IMPORT_PREFIX } from '../../content/consts.js';
+import { shorthash } from '../../runtime/server/shorthash.js';
+import { VALID_INPUT_FORMATS } from '../consts.js';
+
+/**
+ * Resolves an image src from a content file (such as markdown) to an import id.
+ * @param imageSrc The src attribute of an image tag
+ * @param filePath The path to the file that contains the image
+ * @returns The import id of the image, or undefined if it is not a local image
+ */
+export function imageSrcToImportId(imageSrc: string, filePath: string): string | undefined {
+	// If the import is coming from the data store it will have a special prefix to identify it
+	// as an image import. We remove this prefix so that we can resolve the image correctly.
+	if (imageSrc.startsWith(IMAGE_IMPORT_PREFIX)) {
+		imageSrc = imageSrc.slice(IMAGE_IMPORT_PREFIX.length);
+	}
+
+	// We only care about local imports
+	if (['http', 'https'].includes(imageSrc.split(':')[0]) || imageSrc.startsWith('/')) {
```

Comment:
There's a `isRemotePath` util you can use here

https://github.com/withastro/astro/blob/16ed760f99bfd8da05473548ccc15a1a045b238c/packages/internal-helpers/src/path.ts#L88****